File: fizzbuzz.scala
object FizzBuzz {
  def fizzBuzz(n: Int): String = n match {
    case _ if n % 15 == 0 => "FizzBuzz"
    case _ if n % 3 == 0  => "Fizz"
    case _ if n % 5 == 0  => "Buzz"
    case _                => n.toString
  }

  def main(args: Array[String]): Unit = {
    (1 to 100).foreach(i => println(fizzBuzz(i)))
  }
}

File: fizzbuzz.rs
fn fizz_buzz(n: i32) -> String {
    match (n % 3, n % 5) {
        (0, 0) => "FizzBuzz".to_string(),
        (0, _) => "Fizz".to_string(),
        (_, 0) => "Buzz".to_string(),
        _ => n.to_string(),
    }
}

fn main() {
    for i in 1..=100 {
        println!("{}", fizz_buzz(i));
    }
}